Sources/Core/Utilities/Money/Money.swift

```diff
@@ -0,0 +1,291 @@
+      currencyCode: currency.code,
+      precision: precision,
+      locale: locale
+    )
+  }
+  
+  /// Locale object from the current _localeIdentifier_
+  var locale: Locale {
+    .init(identifier: localeIdentifier)
+  }
+  
+  /// Return a new instance with precision trimed down to the safest possible scale.
+  ///
+  ///     let m = Money(amount: 99250, precision: 4)
+  ///     let trimed = m.trimedPrecision()
+  ///     XCAssertEqual(trimed.amount, 9925)
+  ///     XCAssertEqual(trimed.precision, 3)
+  func trimedPrecision() -> Self {
+    var res = self
+    res.trimPrecision()
+    return res
+  }
+  
+  /// Trim the instance down to the safest possible scale.
+  ///
+  ///     let m = Money(amount: 99250, precision: 4)
+  ///     let trimed = m.trimedPrecision()
+  ///     XCAssertEqual(trimed.amount, 9925)
+  ///     XCAssertEqual(trimed.precision, 3)
+  mutating func trimPrecision() {
+    while amount % 10 == 0 {
+      amount /= 10
+      precision -= 1
+    }
+  }
+}
+
+// MARK: - Public Methods - Testing
+public extension Money {
+  var isPositive: Bool {
+    amount >= .zero
+  }
+  
+  var isNegative: Bool {
+    amount < .zero
+  }
+  
+  var isZero: Bool {
+    amount == .zero
+  }
+}
+
+// MARK: - Codable
+extension Money: Codable {
+  private enum CodingKeys: CodingKey {
+    case cents, currency, localeIdentifier, precision
+  }
+  
+  public init(from decoder: Decoder) throws {
+    let values = try decoder.container(keyedBy: CodingKeys.self)
+    amount = try values.decode(Cents.self, forKey: .cents)
+    currency = try values.decode(Currency.self, forKey: .currency)
+    localeIdentifier = try values.decode(String.self, forKey: .localeIdentifier)
+    precision = try values.decode(Int.self, forKey: .precision)
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(amount, forKey: .cents)
+    try container.encode(currency, forKey: .currency)
+    try container.encode(localeIdentifier, forKey: .localeIdentifier)
+    try container.encode(precision, forKey: .precision)
+  }
+}
+
+// MARK: - ExpressibleByFloatLiteral, CustomStringConvertible
+extension Money: ExpressibleByFloatLiteral, CustomStringConvertible {
+  /// Initialize a new Money instance from a literal.
+  ///
+  /// The literal gets automatically converted to Cents.
+  /// We use `ExpressibleByFloatLiteral` instead of
+  /// `ExpressibleByIntegerLiteral` so that multiplying by
+  /// a factor works as expected.
+  public init(floatLiteral cents: Double) {
+    self.amount = Cents(cents)
+    self.currency = defaults.currency
+    self.localeIdentifier = defaults.locale.identifier
+    self.precision = 2
+  }
+  
+  public var description: String {
+    return formatted ?? "\(amount) \(currency.symbol)"
+  }
+}
+
+// MARK: - Math operators
+
+public func + (_ lhs: Money, _ rhs: Money) -> Money {
+  precondition(lhs.currency == rhs.currency)
```

Comment:
I would usually go with the comment, but the user will not see the comments for this method because XCode threat it as a basic operation (the "control + space" does not have any results) :/ 
On the other hand, the throw will add more complexity to the implementation, so I am not sure what is the best solution here.
Maybe to go with the precondition with the comment for now?

Reply:
Ok, lets go with the throw!

Reply:
There is no optimal solution here. Eitherway, the edge cases are crucial here. 

Going with `precondition` to avoid cumbersomeness, the app will crash.
Going with `throw` to avoid crashing, a cumbersome experience.

Let's discuss this in a pod meeting today.

Reply:
1. App crash is not always _bad_. At least it indicates that you made a mistake and can fix it.
2. We shouldn't design libraries in a way that will make it painful for users to use just because some (non-diligent) programmers can use them in an incorrect way.
3. We can swap `precondition` with `assert` so that release builds remove the check. You are still able to catch bugs during development.

Reply:
However, I'm not confident that 3) is a good solution. If for some reason, in production, it just so happens that math is performed on two Money values with a different currency, the bug that will appear will be very subtle but the overall mistake might be much worse than a crash. For instance, if we display an incorrect amount of money to the user and they make an action based on that, that can have big consequences.

Reply:
I would go with `fatalError` here because I would like to see app termination in release builds as well.
```swift
guard lhs.currency == rhs.currency else { fatalError("When adding Money, both currencies must equal!" }
```